bee-block/src/output/mod.rs

```diff
@@ -259,22 +254,79 @@ impl Output {
     }
 }
 
+impl Packable for Output {
+    type UnpackError = Error;
+    type UnpackVisitor = ProtocolParameters;
+
+    fn pack<P: Packer>(&self, packer: &mut P) -> Result<(), P::Error> {
+        match self {
+            Output::Treasury(output) => {
+                TreasuryOutput::KIND.pack(packer)?;
+                output.pack(packer)
+            }
+            Output::Basic(output) => {
+                BasicOutput::KIND.pack(packer)?;
+                output.pack(packer)
+            }
+            Output::Alias(output) => {
+                AliasOutput::KIND.pack(packer)?;
+                output.pack(packer)
+            }
+            Output::Foundry(output) => {
+                FoundryOutput::KIND.pack(packer)?;
+                output.pack(packer)
+            }
+            Output::Nft(output) => {
+                NftOutput::KIND.pack(packer)?;
+                output.pack(packer)
+            }
+        }?;
+
+        Ok(())
+    }
+
+    fn unpack<U: Unpacker, const VERIFY: bool>(
+        unpacker: &mut U,
+        visitor: &Self::UnpackVisitor,
+    ) -> Result<Self, UnpackError<Self::UnpackError, U::Error>> {
+        Ok(match u8::unpack::<_, VERIFY>(unpacker, &()).coerce()? {
+            TreasuryOutput::KIND => Output::from(TreasuryOutput::unpack::<_, VERIFY>(unpacker, &()).coerce()?),
+            BasicOutput::KIND => Output::from(BasicOutput::unpack::<_, VERIFY>(unpacker, visitor).coerce()?),
+            AliasOutput::KIND => Output::from(AliasOutput::unpack::<_, VERIFY>(unpacker, visitor).coerce()?),
+            FoundryOutput::KIND => Output::from(FoundryOutput::unpack::<_, VERIFY>(unpacker, visitor).coerce()?),
+            NftOutput::KIND => Output::from(NftOutput::unpack::<_, VERIFY>(unpacker, visitor).coerce()?),
+            k => return Err(Error::InvalidOutputKind(k)).map_err(UnpackError::Packable),
+        })
+    }
+}
+
 impl Rent for Output {
     fn weighted_bytes(&self, rent_structure: &RentStructure) -> u64 {
         self.packed_len() as u64 * rent_structure.v_byte_factor_data as u64
     }
 }
 
+pub(crate) fn verify_output_amount<const VERIFY: bool>(
+    amount: &u64,
+    protocol_parameters: &ProtocolParameters,
+) -> Result<(), Error> {
+    if VERIFY && (*amount < Output::AMOUNT_MIN || *amount > protocol_parameters.token_supply()) {
```

Comment:
Should AMOUNT_MIN stay hardcoded?

Reply:
I'd say yes as it's part of the TIP https://github.com/iotaledger/tips/blob/main/tips/TIP-0018/tip-0018.md#additional-transaction-syntactic-validation-rules